### PR TITLE
Add automatic setting logic for ENABLE_LEGACY_FSGROUP_INJECTION

### PR DESCRIPTION
**Pull request by kfaseela**
The previous logic was available only for IstioOperator,
hence helm based integration tests on k8s 1.16 were failing.

Signed-off-by: Faseela K <faseela.k@est.tech>

**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [X] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: manifests/charts/istio-control/istio-discovery/templates/deployment.yaml [hunk 1]
```diff
@@ -149,6 +149,10 @@ spec:
             value: "{{ $val }}"
           {{- end }}
           {{- end }}
+{{- if semverCompare "<1.19" .Capabilities.KubeVersion.GitVersion }}
+          - name: ENABLE_LEGACY_FSGROUP_INJECTION
```
Comment (AndreaM12345) [inline manifests/charts/istio-control/istio-discovery/templates/deployment.yaml:153]: does a check for JWT Policy values need to be added too?

Comment (howardjohn) [reply manifests/charts/istio-control/istio-discovery/templates/deployment.yaml:153]: It doesn't _need_ to be false -- it just _can_ be. IMO its not needed; this is strictly an improvement for Helm. While we could add the jwt check in helm as well, the jwt change is from Kubernetes 1.12 which is ~forever ago, so I am not sure its worth the effort

Comment (AndreaM12345) [reply manifests/charts/istio-control/istio-discovery/templates/deployment.yaml:153]: okay thanks for the clarification

**Comment by ericvn** [comment]
Do we need this in release-1.15?

**Comment by kfaseela** [comment]
@ericvn : I would wait till all the post submit tests pass before putting to 1.15

**Pull request merged**